### PR TITLE
[EDITOR UTILS] embed pre-processing + fixed HTML parsing:

### PR DESCRIPTION
**Pull request by jerome-poisson**
pre-processing of embed for customisation is now possible using
callables set in the new `EMBED_PRE_PROCESS` setting.

This patch also fix parsing of HTML, as part of the HTML could be lost
with DOM.parse_html + selecting only children (seen with QUMU embed).
As we use `lxml` as draftjs_exporter backend, we can use
`superdesk.etree.parse_html`.

SDESK-4939

### CODE REVIEW

**Comment by jerome-poisson** [comment]
@petrjasek FYI I've added a test + modified the parsing of HTML in `superdesk.etree` to fix handling of some corner cases (notably the use of `<script>` as it is the case with QUMU).

File: docs/editor.rst [hunk 1]
```diff
@@ -48,3 +48,15 @@ show how to add an embedded code at the top of the body::
...
+Rendering customisation
+=======================
+
+Sometimes, you may want to modify the behavior of the renderer.
+
+embeds
+------
+
+Embeds can be modified before being rendered. For that, the ``EMBED_PRE_PROCESS`` setting
+can be used. It it an iterable linking to callable. Each callable will get the data dict
```
Comment (ride90) [inline docs/editor.rst]: Typo "It it" -> "It is"

File: tests/editor_utils_test.py [hunk 2]
```diff
@@ -11,13 +11,18 @@
...
 from superdesk.editor_utils import Editor3Content
 
 
-class Editor3TestCase(unittest.TestCase):
+class Editor3TestCase(TestCase):
+
+    def setUp(self):
+        super().setUp()
+        if "EMBED_PRE_PROCESS" in self.app.config:
+            del self.app.config["EMBED_PRE_PROCESS"]
```
Comment (ride90) [inline tests/editor_utils_test.py]: why do we need this?

Comment (jerome-poisson) [reply tests/editor_utils_test.py]: Because the config is global to the whole test, and when I modify `EMBED_PRE_PROCESS` at https://github.com/superdesk/superdesk-core/pull/1807/files#diff-80f1d1a029ab2f40ac5996f23c54282aR1614 it will impact all following test, which may fail if they use embed. I'm not sure it there is a cleaner way.

Comment (ride90) [reply tests/editor_utils_test.py]: Got it thanks.
I think `setUp` is called before every test anyway and by default `EMBED_PRE_PROCESS` is not set anyway.

Comment (ride90) [reply tests/editor_utils_test.py]: ahh, ok got it now. You can ignore my previous comment :)

File: docs/editor.rst (hunk 1 above)
Comment (jerome-poisson) [reply docs/editor.rst]: fixed, thx
